addons/web_editor/static/lib/odoo-editor/src/utils/utils.js

```diff
@@ -1429,29 +1682,34 @@ export function splitElement(element, offset) {
  */
 export function splitAroundUntil(elements, limitAncestor) {
     elements = Array.isArray(elements) ? elements : [elements];
-    let after = elements[elements.length - 1].nextSibling;
-    let newUntil = limitAncestor;
+    const firstNode = elements[0];
+    const lastNode = elements[elements.length - 1];
+    if ([firstNode, lastNode].includes(limitAncestor)) {
+        return limitAncestor;
+    }
+    let before = firstNode.previousSibling;
+    let after = lastNode.nextSibling;
     let beforeSplit, afterSplit;
+    if (!before && !after && elements[0] !== limitAncestor) {
+        return splitAroundUntil(elements[0].parentElement, limitAncestor);
+    }
     // Split up ancestors up to font
     while (after && after.parentElement !== limitAncestor) {
         afterSplit = splitElement(after.parentElement, childNodeIndex(after))[0];
-        newUntil = afterSplit;
-        after = newUntil.nextSibling;
+        after = afterSplit.nextSibling;
     }
     if (after) {
         afterSplit = splitElement(limitAncestor, childNodeIndex(after))[0];
         limitAncestor = afterSplit;
     }
-    let before = elements[0].previousSibling;
     while (before && before.parentElement !== limitAncestor) {
         beforeSplit = splitElement(before.parentElement, childNodeIndex(before) + 1)[1];
-        newUntil = beforeSplit;
-        before = newUntil.previousSibling;
+        before = beforeSplit.previousSibling;
     }
     if (before) {
         beforeSplit = splitElement(limitAncestor, childNodeIndex(before) + 1)[1];
     }
-    return [beforeSplit, afterSplit];
+    return beforeSplit || afterSplit || limitAncestor;
```

Comment:
That is quite the change in the util here. I'm curious about it so I'd be interested in discussing this together tomorrow.